### PR TITLE
Support Symfony 3.3

### LINKED ISSUE #8165: On Symfony 3.3.2, problem parsing routing/cart.yml
```
[Symfony\Component\Config\Exception\FileLoaderLoadException]                                                                                                              
  The file "alias: sylius.order                                                                                                                                             
  path: '" does not exist in /var/www/vendor/sylius/sylius/src/Sylius/Bundle/AdminApiBundle/Resources/config/routing/cart.yml (which is being imported from "/var/www/vend  
  or/sylius/sylius/src/Sylius/Bundle/AdminApiBundle/Resources/config/routing/main.yml"). 
```
Am I the only one getting this error ?
If I remove this part, everything works.
```
sylius_admin_api_carts:
    resource: |
        alias: sylius.order
        path: '/'
        only: [index]
        grid: sylius_admin_api_cart
        section: admin_api
        serialization_version: $version
    type: sylius.resource_api
```

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | kinda |
| BC breaks?      | no |
| Related tickets | closes #8165, closes #8247 |
| License         | MIT |

Blocking:
 - https://github.com/symfony/symfony/pull/23411 (workaround https://github.com/Sylius/Sylius/pull/8120/commits/951e26db7f69d4e55a51a5a3cf7dc2f242c3bdd3)
 - https://github.com/minkphp/MinkBrowserKitDriver/pull/118 (workaround https://github.com/Sylius/Sylius/pull/8120/commits/6f7d51fe83167f6cbe2106f34ea6581fca69fbdf)

Nice to have:
 - https://github.com/symfony/symfony/issues/23412 (workaround https://github.com/Sylius/Sylius/pull/8247)

### CODE REVIEW

**Comment by psihius** [comment]
There are issues with just bumping Symfony version - things don't work out of the box #8123

**Comment by wadjeroudi** [comment]
@pamil nice job. It seems good for symfony 3.3 support ? Is there something left ?

File: src/Sylius/Bundle/UserBundle/Authentication/DefaultAuthenticationSuccessHandler.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * TODO: Workaround for regression introduced in Symfony 3.3. To be deleted when fixed.
+ *
+ * @see https://github.com/symfony/symfony/pull/23411
+ * @see https://github.com/symfony/symfony/pull/23061
+ *
+ * @internal
+ *
+ * {@inheritdoc}
+ */
+class DefaultAuthenticationSuccessHandler extends SymfonyDefaultAuthenticationSuccessHandler
```
Comment (lchrusciel) [inline src/Sylius/Bundle/UserBundle/Authentication/DefaultAuthenticationSuccessHandler.php]: Can't we make it final?

**Comment by pjedrzejewski** [comment]
Thank you Kamil, awesome work!

File: src/Sylius/Bundle/UserBundle/Authentication/DefaultAuthenticationSuccessHandler.php (hunk 1 above)
Comment (pamil) [inline src/Sylius/Bundle/UserBundle/Authentication/DefaultAuthenticationSuccessHandler.php]: It's extended by `Sylius\Bundle\UserBundle\Authentication\AuthenticationSuccessHandler` (and internal :)).

**Pull request merged**